jingdai/mailroom.py

```diff
@@ -50,24 +51,53 @@ def create_a_report(donation):
     header_line='{:>12}  |{:>12}  |{:>12}  |{:>12}  '.format("Donor Name","Total Given","Num Gifts","Average Gift")
     print(header_line)
     for sum_list in sum_lists:
-        line_new = '{:>12}  ${:>12}  {:>12}  ${:>12}  '.format(*sum_list)
+        line_new = '{:>12}  ${:>12}  {:>12}  ${:+.2f}  '.format(*sum_list)
         print(line_new)
     return
 
+def send_a_letter(donation):
+    names=[] #create a list which has all the donor's name
+    for pair in donation:
+        if (pair[0],pair[1]) not in names:
+            name=(pair[0],pair[1])
+            names.append(name)
+
+    sum_lists=[] # create a list which has donor's name and the sum donation amount.
+    for i,z in names:
+        sum=0
+        for j in donation:
+            if (i,z)== (j[0],j[1]):
+                sum=j[2]+sum
+                sum_list=(j[0],j[1],sum)
```

Comment:
Think about using descriptive constants to take the place of your integer indexes.  Something like this:
```
FIRST_NAME = 0
LAST_NAME = 1
AMOUNT = 2
```